flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/StreamTableEnvironment.scala

```diff
@@ -667,30 +719,62 @@ abstract class StreamTableEnvironment(
     // get CRow plan
     val plan: DataStream[CRow] = translateToCRow(logicalPlan, queryConfig)
 
+    val rowtimeFields = logicalType
+      .getFieldList.asScala
+      .filter(f => FlinkTypeFactory.isRowtimeIndicatorType(f.getType))
+
+    // convert the input type for the conversion mapper
+    // the input will be changed in the OutputRowtimeProcessFunction later
+    val convType = if (rowtimeFields.size > 1) {
+      throw new TableException(
```

Comment:
What would happen if the left most table does not have a time attribute (or if it is projected out)? I just think that the semantics of the `StreamRecord` timestamps are too important to have an implicit behavior that is hard to explain and reason about for users. IMO, an exception that asks for explicit user input is the better choice compared to a behavior that depends on non-obvious query characteristics and is hard to predict.